openmc/deplete/flux_operator.py

```diff
@@ -0,0 +1,601 @@
+        else:
+            self._normalization_helper = SourceRateHelper()
+
+        # Select and create fission yield helper
+        fission_helper = ConstantFissionYieldHelper
+        self._yield_helper = fission_helper.from_operator(
+            self, **fission_yield_opts)
+
+    def initial_condition(self):
+        """Performs final setup and returns initial condition.
+
+        Returns
+        -------
+        list of numpy.ndarray
+            Total density for initial conditions.
+        """
+
+        # Return number density vector
+        return super().initial_condition(self.materials)
+
+    def __call__(self, vec, source_rate):
+        """Obtain the reaction rates
+
+        Parameters
+        ----------
+        vec : list of numpy.ndarray
+            Total atoms to be used in function.
+        source_rate : float
+            Power in [W] or flux in [neut/s-cm^2]
+
+        Returns
+        -------
+        openmc.deplete.OperatorResult
+            Eigenvalue and reaction rates resulting from transport operator
+
+        """
+
+        self._update_materials_and_nuclides(vec)
+
+        rates = self._calculate_reaction_rates(source_rate)
+        keff = self._keff
+
+        op_result = OperatorResult(keff, rates)
+        return copy.deepcopy(op_result)
+
+    def _update_materials(self):
+        """Updates material compositions in OpenMC on all processes."""
+
+        for rank in range(comm.size):
+            number_i = comm.bcast(self.number, root=rank)
+
+            for mat in number_i.materials:
+                nuclides = []
+                densities = []
+                for nuc in number_i.nuclides:
+                    if nuc in self.nuclides_with_data:
+                        val = 1.0e-24 * number_i.get_atom_density(mat, nuc)
+
+                        # If nuclide is zero, do not add to the problem.
+                        if val > 0.0:
+                            if self.round_number:
+                                val_magnitude = np.floor(np.log10(val))
+                                val_scaled = val / 10**val_magnitude
+                                val_round = round(val_scaled, 8)
+
+                                val = val_round * 10**val_magnitude
+
+                            nuclides.append(nuc)
+                            densities.append(val)
+                        else:
+                            # Only output warnings if values are significantly
+                            # negative. CRAM does not guarantee positive
+                            # values.
+                            if val < -1.0e-21:
+                                print(f'WARNING: nuclide {nuc} in material'
+                                      f'{mat} is negative (density = {val}'
+
+                                      ' atom/b-cm)')
+                            number_i[mat, nuc] = 0.0
+
+    @staticmethod
+    def create_micro_xs_from_data_array(nuclides, reactions, data):
```

Comment:
So now we have:
- `generate_1g_cross_sections`
- `FluxDepletionOperator.create_micro_xs_from_data_array`
- `FluxDepletionOperator.create_micro_xs_from_csv`

I'm wondering if maybe introducing a class to store 1g cross sections (that subclasses pandas.DataFrame) might be worthwhile. In that case, we could have:
```Python
class MicroXS(pandas.DataFrame):
    @classmethod
    def from_model(cls, ...):
        ...

    @classmethod
    def from_array(cls, ...):
        ...

    @classmethod
    def from_csv(cls, ...):
        ...
```
What do you think? I could also envision other uses of this class in the future, e.g., interpolating between multiple sets of micros.

Reply:
I like this idea. I'll add it to the PR. Any idea on where it should go? Probably not in `deplete`...

Reply:
I'd say for now, you can leave it in `deplete` since that's the only immediate application. We can always reconsider where the best place is for it to live later